idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/lineMarker/LineMarkerSyntax.kt

```diff
@@ -244,6 +246,30 @@ interface LineMarkerSyntax {
       }
     )
 
+  /**
+   * Similar to [addLineMarkerProviderM], is an extension for PsiElements that are not leafs
```

Comment:
Maybe we can add here some explanation about how it takes the non leaf node and tries to find a leaf node from it.

Reply:
I'd prefer to move this part to the proofs implementation until there is a general abstraction and not in the algebra.